### PR TITLE
feat(quantic): new Quantic Smart Snippet component created

### PR DESCRIPTION
**Pull request by mmitiche**
[SVCC-180](https://coveord.atlassian.net/browse/SVCC-180)
- The Quantic Smart Snippet component created!

#### Design doc:
[Design-doc__Quantic-Smart-Snippet-components](https://docs.google.com/document/d/1kJswWu6qHp5ExMHQagkrCWphrjRmys8a5tQ0oAnQkJU/edit)

#### Demo:

https://user-images.githubusercontent.com/86681870/216449392-71cd4601-abb3-4037-8407-e1619d34f094.mov

[SVCC-180]: https://coveord.atlassian.net/browse/SVCC-180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/quanticSmartSnippet.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+/** @typedef {import("coveo").SmartSnippet} SmartSnippet */
+/** @typedef {import("coveo").SmartSnippetState} SmartSnippetState */
+
+const expandableAnswerBaseClass = 'smart-snippet__answer slds-is-relative';
+
+/**
+ * The `QuanticSmartSnippet` component displays the excerpt of a document that would be most likely to answer a particular query.
+ * @category Search
+ * @example
+ *  <c-quantic-smart-snippet maximum-snippet-height="250"></c-quantic-smart-snippet>
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/quanticSmartSnippet.js:21]:
```suggestion
 *  <c-quantic-smart-snippet engine-id={engineId} maximum-snippet-height="250"></c-quantic-smart-snippet>
```

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/quanticSmartSnippet.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    this.state = this.smartSnippet.state;
+    if (this.answer !== this.state.answer) {
+      this.answer = this.state.answer;
+      this.initializeExpandableAnswerClass();
+      this.expandableAnswerDisplayUpdated = false;
+    }
+  }
+
+  /**
+   * Updates the display of the expandable smart snippet answer.c/exampleSearch
```
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/quanticSmartSnippet.js:94]:
```suggestion
   * Updates the display of the expandable smart snippet answer.
```

**Review by lbergeron — APPROVED** [review]
It looks great! Good job!

**Review by SimonMilord — APPROVED** [review]
GG Mehdi!

**Pull request merged**